src/main/scala/freeeventsourcing/syntax/ProcessSyntax.scala

```diff
@@ -225,9 +234,27 @@ class ProcessSyntax[BC <: BoundedContext](boundedContext: BC) {
 
       final class FromAggregateBuilder[A <: Aggregate] private[FirstOfBuilder] (aggregateType: A, aggregate: A#Id) {
         /** Await an event from the aggregate. */
-        def on[E <: A#Event: AggregateEventType[A, ?]: Typeable](implicit ev: ValidAggregate[BC, A], s: StringSerializable[A#Id]) = {
+        def on[E <: A#Event: AggregateEventType[A, ?]: Typeable](implicit ev: ValidAggregate[BC, A], s: StringSerializable[A#Id]) =
+          when[E].select
+
+        /** Advanced selection of an element (filter, after). */
+        def when[E <: A#Event: AggregateEventType[A, ?]: Typeable](implicit ev: ValidAggregate[BC, A], s: StringSerializable[A#Id]) = {
           val selector = AggregateEventSelector(aggregateType)(aggregate)[E]
-          FirstOfBuilder.this.on(selector)
+          new SelectorBuilder(selector)
+        }
+        //TODO add a test
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Take the required action to fix the issue indicated by this comment.](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2672543678/issues/source?bid=3298199&fileBranchId=3311368#l245)